django_filebased_email_backend_ng/backend.py

```diff
@@ -10,20 +12,31 @@
 
 class EmailBackend(BaseEmailBackend):
     def send_messages(self, messages):
-        try:
-            shutil.rmtree(settings.EMAIL_FILE_PATH)
-        except OSError:
-            pass
+        base_dir = settings.EMAIL_FILE_PATH
 
-        os.makedirs(settings.EMAIL_FILE_PATH)
+        # Files will be stored in a dir which name is calculated
+        # with inspection of where this method is called from.
+        #
+        # Knowing that a same function calling send_mail() multiple times
+        # will have the same frame stack,
+        # the emails sent by a same usage will be aggregated in a unique folder
+        usage_aggregation_dir = hashlib.sha1(str(
+            inspect.getouterframes(inspect.currentframe())
+        ).encode('utf-8')).hexdigest()
 
-        for idx, message in enumerate(messages):
-            base = os.path.join(settings.EMAIL_FILE_PATH, '%s' % idx)
+        # we can now safely delete everything in base dir
+        # except usage aggregation folder
+        for d in os.listdir(base_dir):
+            if d != usage_aggregation_dir:
+                shutil.rmtree(os.path.join(base_dir, d), ignore_errors=True)
 
-            os.makedirs(base)
+        for message in messages:
+            message_dir = '__'.join(message.to).replace('@', '_AT_')
```

Comment:
There are surely other characters that would need replacing. (Why don't we whitelist instead of blacklist?)